src/main/java/util/Utility.java

```diff
@@ -190,17 +187,16 @@ public static int safeLongToInt(long l) {
     public static Date parseHTTPLastModifiedDate(String dateString) {
         assert dateString != null;
         try {
-            return new SimpleDateFormat("EEE, d MMM yyyy HH:mm:ss Z").parse(dateString);
+            return new SimpleDateFormat("EEE, d MMM yyyy HH:mm:ss Z", Locale.US).parse(dateString);
         } catch (ParseException e) {
-            assert false : "Error in date format string!";
+            logger.error(e.getLocalizedMessage(), e);
+            throw new IllegalArgumentException("Could not parse date " + dateString);
```

Comment:
It can't be handled there (without defaulting to some date, which I'm not sure we want). Instead we log the exception, then let it propagate back up to RepoIO, where it will be handled [here](https://github.com/HubTurbo/HubTurbo/blob/master/src/main/java/backend/RepoIO.java#L160).